docs/advanced_documentation/build-guide.md

```diff
@@ -100,40 +100,54 @@ Once you have prepared the build dependencies,
 you can install the library from source in develop mode with the development dependency. 
 Go to the root folder of the repository.
 
-```
+```shell
 pip install -e .[dev]
 ```
 
 Then you can run the tests.
 
-```
+```shell
 pytest
 ```
 
 A basic `self_test` function is provided to check if the installation was successful and ensures there are no build errors, segmentation violations, undefined symbols, etc. It performs multiple C API calls, runs through the main data flow, and verifies the integrity of serialization and deserialization.
 
-```
+```python
 from power_grid_model.utils import self_test
 self_test()
 ```
 
 ## Build CMake Project
 
-There is a root cmake file in the root folder of the repo `CMakeLists.txt`. It specifies
-dependencies and the build options for the project. The core algorithm is implemented in the header-only
-library `power_grid_model`. There are four sub-projects defined in the root cmake file:
+### User build
+
+If you are a C-API user of the library, you can build the CMake using all the default settings.
+You can specifiy a standard [CMAKE_BUILD_TYPE](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html).
+This will only build the core C-API dynamic library.
+
+```shell
+cmake -DCMAKE_BUILD_TYPE=Release -B build
+cmake --build build --config Release
+```
+
+### Developer build
+
+If you are a developer of Power Grid Model, 
```

Comment:
```suggestion
If you opt for a developer build of Power Grid Model, 
```
This may not necessarily be true? A developer of PGM sounds like officially affiliated titles ;)